File: lib/data/sources/local/daos/chapter_dao.dart
import 'package:froom/froom.dart';

import '../../../entities/chapter_entity.dart';

@dao
abstract class ChapterDao {
  @Query(
    'SELECT * FROM chapters WHERE bibleId = :bibleId AND bookId = :bookId ORDER BY CAST(number AS INTEGER)',
  )
  Future<List<ChapterEntity>> getChaptersByBook(
    String bibleId,
    String bookId,
  );

  @Query('SELECT * FROM chapters WHERE id = :chapterId AND bibleId = :bibleId')
  Future<ChapterEntity?> getChapter(String chapterId, String bibleId);

  @Insert(onConflict: OnConflictStrategy.replace)
  Future<void> insertChapters(List<ChapterEntity> chapters);

  @Query('DELETE FROM chapters WHERE bibleId = :bibleId')
  Future<void> deleteChaptersByBible(String bibleId);
}

File: core/database/src/main/java/com/biblelib/core/database/daos/ChapterDao.kt
package com.biblelib.core.database.daos

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.biblelib.core.database.model.ChapterEntity

@Dao
interface ChapterDao {
    @Query("SELECT * FROM chapters WHERE bibleAbbr = :abbr AND bookId = :bookId ORDER BY CAST(number AS INTEGER) ASC")
    suspend fun getByBook(abbr: String, bookId: String): List<ChapterEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(records: List<ChapterEntity>)

    @Query("DELETE FROM chapters WHERE bibleAbbr = :abbr")
    suspend fun deleteByBible(abbr: String)

    @Query("DELETE FROM chapters")
    suspend fun deleteAll()
}
